### PR TITLE
Allows using the github-resolver without a PAT

### PR DESCRIPTION
**Pull request by regismesquita**
Allows using the github-resolver without a PAT

### CODE REVIEW

**Comment by neubig** [comment]
Hey @regismesquita , thanks for this! Quick question, after this PR will PRs sent to this repo still come from `openhands-agent`? I'd like to preserve that possibility, even if it's optional.

**Review by neubig — CHANGES_REQUESTED** [review]
OK, could we please change this so that when PAT_TOKEN is set, it uses PAT_TOKEN, and if not it uses the github token? That will maintain the current behavior if PAT_TOKEN is set.

@openhands-agent could you take a first stab at this?

**Comment by enyst** [comment]
> OK, could we please change this so that when PAT_TOKEN is set, it uses PAT_TOKEN, and if not it uses the github token? That will maintain the current behavior if PAT_TOKEN is set.
> 
> @openhands-agent could you take a first stab at this?

I'm not sure if the call in the body of a comment is on pypi yet... @openhands-agent-exp you have been called to help, can you solve the PAT_TOKEN as pointed out above?

**Comment by regismesquita** [comment]
@neubig if the PAT_TOKEN is set it is to be used, I was referring to when using the github token.

**Review by neubig — COMMENTED** [review]
Looks good to me, @malhotra5 could you check too?

File: openhands/resolver/README.md [hunk 1]
```diff
@@ -32,11 +32,11 @@ Follow these steps to use this workflow in your own repository:
...
 5. Set up [GitHub secrets](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions):
    - Required:
+    - `LLM_API_KEY`: Your LLM API key
+   - Optional:
      - `PAT_USERNAME`: GitHub username for the personal access token
      - `PAT_TOKEN`: The personal access token
-     - `LLM_MODEL`: LLM model to use (e.g., "anthropic/claude-3-5-sonnet-20241022")
-     - `LLM_API_KEY`: Your LLM API key
-   - Optional:
+     - `LLM_MODEL`: LLM model to use (defaults to "anthropic/claude-3-5-sonnet-20241022")
```
Comment (malhotra5) [inline openhands/resolver/README.md:39]: `LLM_MODEL` is no longer supported as a secret as per this PR. We should add the specification in the [docs](https://github.com/All-Hands-AI/OpenHands/blob/main/docs/modules/usage/how-to/github-action.md#custom-configurations)
```suggestion
```

**Review by malhotra5 — APPROVED** [review]
I've left a comment for documenting the changes; otherwise LGTM!

File: .github/workflows/openhands-resolver.yml [hunk 2]
```diff
@@ -106,20 +110,34 @@ jobs:
 
       - name: Check required environment variables
         env:
-          LLM_MODEL: ${{ secrets.LLM_MODEL }}
+          LLM_MODEL: ${{ secrets.LLM_MODEL || inputs.LLM_MODEL }}
           LLM_API_KEY: ${{ secrets.LLM_API_KEY }}
           LLM_BASE_URL: ${{ secrets.LLM_BASE_URL }}
           PAT_TOKEN: ${{ secrets.PAT_TOKEN }}
           PAT_USERNAME: ${{ secrets.PAT_USERNAME }}
+          GITHUB_TOKEN: ${{ github.token }}
```
Comment (enyst) [inline .github/workflows/openhands-resolver.yml:118]: Sorry, this is probably obvious and I'm missing something, could you please tell where or how is this `github.token` set to be available to the workflow here?

Comment (regismesquita) [reply .github/workflows/openhands-resolver.yml:118]: It is added by github itself.
https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication

**Review by neubig — APPROVED** [review]
LGTM too, thanks!

File: .github/workflows/openhands-resolver.yml (hunk 2 above)
Comment (enyst) [reply .github/workflows/openhands-resolver.yml:118]: Ah, thanks!

**Pull request merged**